### ISSUE TITLE
Consider adding expm1 and log1p

### ISSUE THREAD

**Issue by NeilGirdhar** [tensorflow/tensorflow#3682]
`expm1` is useful for inverting `tf.nn.softplus`:

```
def inverse_softplus(x):
    return tf.select(tf.gt(x, 80.0),
                     x,
                     tf.log(tf.expm1(x)))
```

`log1p` is useful for implementing `logaddexp`:

```
def logaddexp(x, y):
    temp = x - y
    return tf.select(tf.gt(temp, 0.0),
                     x + tf.log1p(tf.exp(-temp)),
                     y + tf.log1p(tf.exp(temp)))
```

**Comment by girving** [comment]
Yep, this would both be nice to have.

**Comment by ibab** [comment]
I've had a quick look at Eigen to see whether these two are already supported as functors.
`log1p` is there, but `expm1` is missing.
That means a good start for `expm1` would be to add it to Eigen.
I think some extra work in Eigen is also needed to get `log1p` to work with GPU ops (e.g. adding it to `Eigen/src/Core/arch/CUDA/MathFunctions.h`)

**Comment by kbrems** [comment]
As a potential alternative, we have implemented a python-only framework for adding custom operators to tensorflow, the Operator Vectorization Library: 
https://github.com/opveclib/opveclib. Multi-threaded C++ and CUDA code is automatically generated from the python operators, which can be used in a Tensorflow application with only a binary install of tensorflow. We have implemented the log1p and expm1 operators as an example. See documentation at http://opveclib.readthedocs.io/en/latest/expm1.html
